util/http/log/redact_http_query_values.go

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package log
+
+import (
+	"errors"
+	"regexp"
+)
+
+const (
+	// httpQueryRegexPattern will match on the HTTP query component and group
+	// HTTP query symbol, parameters, and values.
+	//
+	// e.g. GET https://github.com/awslabs/soci-snapshotter/releases?tag=v0.4.0
+	//
+	// Match 1: ?tag=v0.4.0
+	// Group 1: ?
+	// Group 2: tag
+	// Group 3: v0.4.0
+	httpQueryRegexPattern = `(\?|\&)([^=]+)\=([^&]+)`
+
+	// redactedReplacementString is the pattern used to redact HTTP query values.
+	//
+	// Applying the replacement string to the example above woud result in the following string:
+	//
+	// GET https://github.com/awslabs/soci-snapshotter/releases?tag=<redacted>
+	redactedReplacementString = "$1$2=<redacted>"
+)
+
+var (
+	// httpQueryRegex is a compiled regular expression of httpQueryRegexPattern.
+	httpQueryRegex = regexp.MustCompile(httpQueryRegexPattern)
+)
+
+// RedactHTTPQueryValues parses an error and replaces any occurrences of an HTTP Query
+// value with '<redacted>' to prevent leakage of sensitive information like credentials.
+func RedactHTTPQueryValues(err error) error {
```

Comment:
I think we should do this in a more structured way by parsing the error as a `url.Error`

e.g. https://go.dev/play/p/LuQg7-Kl8mK

Reply:
LGTM